### ISSUE TITLE
Create performance predictor

### ISSUE THREAD

**Issue by neubig** [nightingal3/llm-pretraining-behaviours#48]
Performance prediction predicts model performance on particular pieces of data based on metadata of the model or training setting.

Some example previous papers include [LangRank](https://aclanthology.org/P19-1301/) ([repo](https://github.com/neulab/langrank)) and [NLPerf](https://arxiv.org/abs/2005.00870) ([repo](https://github.com/xiamengzhou/NLPerf)). Separately from these learning-based approaches, there is also work on neural [scaling laws](https://arxiv.org/abs/2001.08361) for language models which try to predict performance based mostly on model size and number of training iterations.

To start implementing performance prediction in this repo, we need, at a minimum:
1. model metadata for some datasets (done in https://github.com/nightingal3/llm-pretraining-behaviours/issues/29)
2. accuracies on various datasets (pending in https://github.com/nightingal3/llm-pretraining-behaviours/issues/46)

Once we have this, we can:
- [x] Write a script that predicts various dataset's accuracy scores, e.g. using [xgboost](https://xgboost.readthedocs.io/en/stable/python/python_api.html)

- [x] Evaluate how well the predictor does when given particular features. For example, a
  - a "scaling law" style model could take in only info about the model size and length of training
  - a "full model metadata" predictor could take in all of the data about the model architecture and training parameters but no features of the data (including the features included in the "scaling law" model
  - a "model and data metadata" predictor could additionally take in metadata about datasets (created in #44)
 
 Here is an example from the LangRank paper of a table that we could create:
![Screenshot 2024-03-12 at 4 39 31 PM](https://github.com/nightingal3/llm-pretraining-behaviours/assets/398875/841543a7-3dc8-46ab-a9ad-5a0fbd015ff9)

- [x] Calculate feature importances, which would tell us which features are the most predictive (in xgboost, this can be calculated based on the number of decision trees that use a particular feature)

Here is an example:
![Screenshot 2024-03-12 at 4 43 17 PM](https://github.com/nightingal3/llm-pretraining-behaviours/assets/398875/272d0db8-f2f8-4c6c-a74d-0193f930e4af)
